text/0143-versioning-and-stability.md

```diff
@@ -0,0 +1,128 @@
+# OpenTelemetry versioning and stability
+
+OpenTelemetry is a large project with strict compatibility requirements. This proposal defines the stability gurantees offered by OpenTelemetry, along with a versioning and lifecycle proposal which defines how we meet those requirements.
+
+Language implementations are expected to follow this proposal exactly, unless a langague or package manager convention interferes significantly. Implementations must take this corss-language proposal, and produce a language-specific proposal which details how these requirements will be met.
+
+## Design goals
+
+**Provide end users with coherent, tested releases.**  
+We do not want users to run untested combinations of core packages. Users must be able to say “I am using OpenTelemetry v1.2.0 for Python” and have a shared understanding as to what that means.
```

Comment:
TBH, I'm not sure what this means, even with this proposal. Is it the API version of a library? Is it an SDK version running in an application? Without that context, I think this statement will always be ambiguous.

Reply:
good point. Would saying that "the release number is the *highest supported* version of the SDK the API" clarify this? That is the intent, at least.

Reply:
I don't grok that. Maybe provide the whole sentence?  

I think if you just put the word "API" or "SDK" in between "OpenTelemetry" and the version, it makes sense. 

The more I think about this, the more I don't understand what you're getting at, TBH. :)

Reply:
Ok, well maybe this is just a distraction and I should delete it. My main point is that all core components should be released together: Don't release a new version of the API without SDK and contrib support; release everything together. But this is a poor way of wording it.

Reply:
I also am curious about the thought behind this.  If one is saying "I am using OpenTelemetry v1.2.0 for Python", that would most likely correlate to a tagged release?  While I agree that a version number <-> code understanding is important, I don't know whether our version numbering is prudent in this example.

Reply:
Yeah, honestly what I was trying to say isn't really clear or necessary, so I'll remove this.